test/test_base_alloc.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#include <assert.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+
+#include "base_alloc.h"
+
+#define NTHREADS 10
+#define ITERATIONS 1000
+#define ALLOCATION_SIZE 16
+
+static void *start_routine(void *arg) {
+    unsigned char *ptrs[ITERATIONS];
+    umf_ba_pool_t *pool = (umf_ba_pool_t *)arg;
+
+    long TID = syscall(SYS_gettid);
+
+    for (int i = 0; i < ITERATIONS; i++) {
+        ptrs[i] = umf_ba_alloc(pool);
+        memset(ptrs[i], (i + TID) & 0xFF, ALLOCATION_SIZE);
+    }
+
+    for (int i = 0; i < ITERATIONS; i++) {
+        for (int k = 0; k < ALLOCATION_SIZE; k++) {
+            if (*(ptrs[i] + k) != ((i + TID) & 0xFF)) {
+                fprintf(stderr,
+                        "i = %i k = %i, *(ptrs[i] + k) = %i != ((i + TID) & "
+                        "0xFF) = %li\n",
+                        i, k, *(ptrs[i] + k), ((i + TID) & 0xFF));
+            }
+            assert(*(ptrs[i] + k) == ((i + TID) & 0xFF));
+        }
+        umf_ba_free(pool, ptrs[i]);
+    }
+
+    return NULL;
+}
+
+int main() {
+    pthread_t thread[NTHREADS];
+    umf_ba_pool_t *pool = umf_ba_create(ALLOCATION_SIZE);
+
+    for (int i = 0; i < NTHREADS; i++) {
+        int ret = pthread_create(&thread[i], NULL, start_routine, pool);
+        if (ret) {
```

Comment:
shouldn't you use `ASSERT_*` macros...?

Reply:
Done

Reply:
@lukaszstolarczuk please resolve

Reply:
heh, no need to ping me twice on the same issue :wink: